test/cpp/end2end/xds_end2end_test.cc

```diff
@@ -5442,16 +5443,36 @@ class XdsSecurityTest : public BasicTest {
       StartBackend(0);
       ResetBackendCounters();
       if (test_expects_failure) {
-        if (!SendRpc().ok()) break;
+        Status status = SendRpc();
+        if (status.ok()) {
+          gpr_log(GPR_ERROR, "RPC succeeded. Failure expected. Trying again.");
+          continue;
+        }
       } else {
         WaitForBackend(0);
-        if (SendRpc().ok() &&
-            backends_[0]->backend_service()->request_count() == 1UL &&
-            backends_[0]->backend_service()->last_peer_identity() ==
-                expected_authenticated_identity) {
-          break;
+        Status status = SendRpc();
+        if (!status.ok()) {
+          gpr_log(GPR_ERROR, "RPC failed. code=%d message=%s Trying again.",
+                  status.error_code(), status.error_message().c_str());
+          continue;
+        }
+        if (backends_[0]->backend_service()->last_peer_identity() !=
+            expected_authenticated_identity) {
+          gpr_log(
+              GPR_ERROR,
+              "Expected client identity does not match. (actual) %s vs "
+              "(expected) %s Trying again.",
+              absl::StrJoin(
+                  backends_[0]->backend_service()->last_peer_identity(), ",")
+                  .c_str(),
+              absl::StrJoin(expected_authenticated_identity, ",").c_str());
+          // It is possible that the new subchannel list has not been subbed in
+          // yet. It will be subbed in once the subchannel connects.
+          gpr_sleep_until(grpc_timeout_milliseconds_to_deadline(2000));
```

Comment:
If I'm understanding the problem correctly, the issue is that the subchannels from the previous test are still in the global subchannel pool when the next test starts, so we're getting the old subchannels instead of creating a brand new set.  Is that right?  If so, I think a better way to address this would be to have the channel use the local subchannel pool instead of the global subchannel pool, so that each test will start by creating its own new subchannels.

Reply:
As discussed, I removed this sleep and instead removed the backends restarting code while increasing the number of retries. Since, we are no longer restarting backends, there is no longer a need to change the channel arguments either.